### PR TITLE
check if import collides with exported local name

### LINKED ISSUE #7195: import and export function name confliction should be an error
**TypeScript Version:** 

master

**Code**

```ts
// lib.ts
export function f() {
}
// index.ts
import {f} from './lib';
export function f() {
}
```

**Expected behavior:**

error

**Actual behavior:** 

pass

### PR DESCRIPTION
**Pull request by vladima**
fixes #7195

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -15063,10 +15063,16 @@ namespace ts {
             const symbol = getSymbolOfNode(node);
             const target = resolveAlias(symbol);
             if (target !== unknownSymbol) {
+                // For external modules symbol represent local symbol for an alias. 
+                // This local symbol will merge any other local declarations (excluding other aliases)
+                // and symbol.flags will contains combined representation for all merged declaration.
+                // Based on symbol.flags we can compute a set of excluded meanings (meaning that resolved alias should not have,
+                // otherwise it will conflict with some local declaration). Note that in addition to normal flags we include matching SymbolFlags.Export* 
+                // in order to prevent collisions with declarations that were exported from the current module (they still contribute to local names).
```
Comment (sandersn) [inline src/compiler/checker.ts]: This means that previously we would correctly catch unexported functions, right? Like:

```ts
import {f} from "f1"
function f() { }
```

Comment (vladima) [inline src/compiler/checker.ts]: yes

**Comment by sandersn** [comment]
:+1:

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**